modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/form_portlet.js

```diff
@@ -790,15 +790,10 @@ AUI.add(
 					},
 
 					_isSameState: function(state1, state2) {
-						var instance = this;
+						state1 = this._removePropertyDeep(state1, 'instanceId');
+						state2 = this._removePropertyDeep(state2, 'instanceId');
 
-						return AUI._.isEqual(
-							state1,
-							state2,
-							function(value1, value2, key) {
-								return (key === 'instanceId') || undefined;
-							}
-						);
+						return JSON.stringify(state1) === JSON.stringify(state2);
```

Comment:
hi @iiventura 

I see a couple problems in this:

- These states can be very big (imagine 15KB). We are building two JSON objects (it's consuming), then stringfying (consuming too) and then comparing this. We are worried about the perfomance of those operations.
- Sometimes, removing a property may make the property order change. We could have an incorrect response of this method because of that.
- Also, this algorithm always works in the worst case. Imagine the first property of the context is the one which is different. The algorithm would go to all the consuming steps anyway.

Could you go over those points and resend? 

Thanks! :)